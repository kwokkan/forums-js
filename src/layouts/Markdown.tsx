import ReactMarkdown from "react-markdown";

interface IProps {
    content: string;
}

export function Markdown(props: IProps) {
    return (
        <ReactMarkdown>
            {props.content}
        </ReactMarkdown>
    );
}
