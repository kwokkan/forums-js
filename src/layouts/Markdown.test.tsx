import { create } from "react-test-renderer";
import { Markdown } from "./Markdown";

test("Renders without errors", () => {
    const content = [
        "# header",
        "## sub header",
        "<script>alert(1)</script>",
        "<b>bold</b>",
        "[my link](http://localhost:3000)",
        "normal content"
    ];
    const tree = create(
        <Markdown content={content.join("\n\n")} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
