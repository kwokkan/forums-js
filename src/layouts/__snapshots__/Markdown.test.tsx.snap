// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without errors 1`] = `
Array [
  <h1>
    header
  </h1>,
  "
",
  <h2>
    sub header
  </h2>,
  "
",
  "<script>alert(1)</script>",
  "
",
  <p>
    &lt;b&gt;
    bold
    &lt;/b&gt;
  </p>,
  "
",
  <p>
    <a
      href="http://localhost:3000"
    >
      my link
    </a>
  </p>,
  "
",
  <p>
    normal content
  </p>,
]
`;
